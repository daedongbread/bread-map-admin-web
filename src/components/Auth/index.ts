export { PersistLogin } from './PersistLogin';
export { RequireAuth } from './RequireAuth';
