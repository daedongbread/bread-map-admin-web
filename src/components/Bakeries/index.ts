export * from './BakeriesTable';
export * from './BakeryFilter';
export * from './BakeryTotalCount';
export * from './BakeryInfoAndFilter';
