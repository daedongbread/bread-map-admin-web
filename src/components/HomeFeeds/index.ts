export * from './HomeFeedsTable';
export * from './FeedSearchBar';
