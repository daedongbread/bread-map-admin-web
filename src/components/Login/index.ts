export * from './Logo';
export * from './LoginForm';
