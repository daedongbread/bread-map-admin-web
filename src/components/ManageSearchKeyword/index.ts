export * from './ManageSearchKeywordCountTable';
export * from './ManageSearchKeywordRankTable';
