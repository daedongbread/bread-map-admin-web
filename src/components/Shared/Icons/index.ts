export * from './Search';
export * from './Pencil';
export * from './BarGraph';
export * from './ChevronDown';
export * from './ChevronLeft';
export * from './ChevronRight';
export * from './Frown';
export * from './Users';
export * from './User';
export * from './Logout';
export * from './Circle';
export * from './CheckLine';
export * from './Download';
export * from './Trash';
export * from './NewBadge';
export * from './Meal';
export * from './Dislike';
export * from './Profile';
export * from './Parking';
export * from './Wifi';
export * from './Pet';
export * from './Delivery';
export * from './Shipping';
export * from './Booking';
export * from './Star';
