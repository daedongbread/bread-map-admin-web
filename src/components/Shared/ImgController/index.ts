export * from './ImgManager';
export * from './ImgSelector';
