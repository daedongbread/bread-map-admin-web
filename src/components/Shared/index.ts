export * from './Header';
export * from './Input';
export * from './Button';
export * from './Pagination';
export * from './Preview';
export * from './SearchBar';
export * from './SelectBox';
export * from './Table';
export * from './Loading';
export * from './Error';
export * from './Tab';
export * from './ImgController';
