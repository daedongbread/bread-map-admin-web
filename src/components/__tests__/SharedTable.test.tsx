import React from 'react';
import { Route } from 'react-router-dom';
import { StatusCell, Table } from '@/components/Shared';
import { BAKERY_STATUS_OPTIONS, BAKERY_TABLE_HEADERS, PATH } from '@/constants';
import { withAllContexts, withRouter, fakeBakeries, fakeEvent } from '@/tests';
import { formatTextToOptionObj } from '@/utils/common';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Shared Table', () => {
  it('이벤트가 없는 테이블 컴포넌트가 렌더링된다', () => {
    const { asFragment } = renderTable({ haveEvent: false });
    expect(asFragment()).toMatchSnapshot();
  });

  it('이벤트가 있는 테이블 컴포넌트가 렌더링된다', () => {
    const { asFragment } = renderTable({ haveEvent: true });
    expect(asFragment()).toMatchSnapshot();
  });

  it('특정 빵집을 누르면 상세페이지로 이동한다', async () => {
    renderTable({ haveEvent: true });

    const target = screen.getAllByTestId('row');
    await userEvent.click(target[0]);
    await waitFor(() => {
      expect(screen.getByText(`상세페이지 베이커리명: ${fakeBakeries[0].name}`)).toBeInTheDocument();
    });
  });

  function renderTable({ haveEvent }: { haveEvent: boolean }) {
    let event = {};
    if (haveEvent) {
      event = fakeEvent;
    }

    const fakeBakeryRow = fakeBakeries.map(item => {
      const status = formatTextToOptionObj({ constants: BAKERY_STATUS_OPTIONS, targetText: item.status });
      return {
        ...item,
        status: <StatusCell color={status.color} text={status.text} />,
      };
    });

    return render(
      withAllContexts(
        withRouter(
          <>
            <Route path="/" element={<Table headers={BAKERY_TABLE_HEADERS} rows={fakeBakeryRow} event={event} />} />
            <Route path={`${PATH.Bakeries}/${fakeBakeries[0].bakeryId}`} element={<p>{`상세페이지 베이커리명: ${fakeBakeries[0].name}`}</p>} />
          </>
        )
      )
    );
  }
});
