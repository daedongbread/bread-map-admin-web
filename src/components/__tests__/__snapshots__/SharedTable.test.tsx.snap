// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Table 이벤트가 없는 테이블 컴포넌트가 렌더링된다 1`] = `
<DocumentFragment>
  <div
    class="css-16omkrh"
  >
    <table
      class="css-gi96c8"
    >
      <thead>
        <tr
          class="css-19bh3p5"
        >
          <th>
            빵집 번호
          </th>
          <th>
            빵집 이름
          </th>
          <th>
            등록일
          </th>
          <th>
            마지막 수정일
          </th>
          <th>
            상태
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="css-19bh3p5"
          data-testid="row"
        >
          <td>
            3
          </td>
          <td>
            테스트 성수점
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            2023-01-16
          </td>
          <td>
            <div
              class="css-1gzqjqd"
              color="#F3213B"
            >
              <div>
                <svg
                  fill="none"
                  fillcolor="#F3213B"
                  height="10"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="3.5"
                    cy="3.5"
                    fill="#F3213B"
                    r="3.5"
                  />
                </svg>
              </div>
              <span>
                미게시
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="css-19bh3p5"
          data-testid="row"
        >
          <td>
            2
          </td>
          <td>
            빵빵 강남점
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            <div
              class="css-1uzinym"
              color="#1EC780"
            >
              <div>
                <svg
                  fill="none"
                  fillcolor="#1EC780"
                  height="10"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="3.5"
                    cy="3.5"
                    fill="#1EC780"
                    r="3.5"
                  />
                </svg>
              </div>
              <span>
                게시중
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="css-19bh3p5"
          data-testid="row"
        >
          <td>
            1
          </td>
          <td>
            일산 브레드
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            <div
              class="css-1gzqjqd"
              color="#F3213B"
            >
              <div>
                <svg
                  fill="none"
                  fillcolor="#F3213B"
                  height="10"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="3.5"
                    cy="3.5"
                    fill="#F3213B"
                    r="3.5"
                  />
                </svg>
              </div>
              <span>
                미게시
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Shared Table 이벤트가 있는 테이블 컴포넌트가 렌더링된다 1`] = `
<DocumentFragment>
  <div
    class="css-16omkrh"
  >
    <table
      class="css-em288z"
    >
      <thead>
        <tr
          class="css-19bh3p5"
        >
          <th>
            빵집 번호
          </th>
          <th>
            빵집 이름
          </th>
          <th>
            등록일
          </th>
          <th>
            마지막 수정일
          </th>
          <th>
            상태
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="css-19bh3p5"
          data-testid="row"
        >
          <td>
            3
          </td>
          <td>
            테스트 성수점
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            2023-01-16
          </td>
          <td>
            <div
              class="css-1gzqjqd"
              color="#F3213B"
            >
              <div>
                <svg
                  fill="none"
                  fillcolor="#F3213B"
                  height="10"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="3.5"
                    cy="3.5"
                    fill="#F3213B"
                    r="3.5"
                  />
                </svg>
              </div>
              <span>
                미게시
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="css-19bh3p5"
          data-testid="row"
        >
          <td>
            2
          </td>
          <td>
            빵빵 강남점
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            <div
              class="css-1uzinym"
              color="#1EC780"
            >
              <div>
                <svg
                  fill="none"
                  fillcolor="#1EC780"
                  height="10"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="3.5"
                    cy="3.5"
                    fill="#1EC780"
                    r="3.5"
                  />
                </svg>
              </div>
              <span>
                게시중
              </span>
            </div>
          </td>
        </tr>
        <tr
          class="css-19bh3p5"
          data-testid="row"
        >
          <td>
            1
          </td>
          <td>
            일산 브레드
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            2023-01-15
          </td>
          <td>
            <div
              class="css-1gzqjqd"
              color="#F3213B"
            >
              <div>
                <svg
                  fill="none"
                  fillcolor="#F3213B"
                  height="10"
                  width="10"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="3.5"
                    cy="3.5"
                    fill="#F3213B"
                    r="3.5"
                  />
                </svg>
              </div>
              <span>
                미게시
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
